### PR TITLE
Docs: Email and Magic Link

### LINKED ISSUE #3849: Docs and guides barely mention normal email-login
# Improve documentation

A page like [this one](https://supabase.io/docs/guides/auth/intro), and [this one](https://supabase.io/docs/guides/auth) should probably have a section about email-auth but I couldn't find but a small mention. Some pages describe an email-link-login, but not really a typical email-auth, besides [this one](https://supabase.io/auth), a little bit.

I feel like docs / guides could include some detailed sections on Email & password auth

I guess there's [this](https://supabase.io/docs/reference/javascript/auth-signup), but other auth seems so prominently mentioned in those other areas. Maybe a link here

### PR DESCRIPTION
**Pull request by Isaiah-Hamilton**
## What kind of change does this PR introduce?

Added requested docs from issue #3849, and will close #3849.

## Additional context

I didn't know where to put the new docs, so let me know If I did anything wrong or needs to be changed.

### CODE REVIEW

File: web/docs/guides/auth/auth-email.mdx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+- Click `Settings` from the list to go to the `Authentication Settings` page
+- Enter the final (hosted) URL of your app under `Site URL` (this is important)
+- Under `Email Auth` turn `Enable Email Signup` to ON
+- Click `Save`
+
+### Add login code to your client app
+
+The JavaScript client code is documented here: [Supabase OAuth Client Code](/docs/reference/javascript/auth-signin#sign-in-using-third-party-providers)
+
+```js
```
Comment (kiwicopple) [inline web/docs/guides/auth/auth-email.mdx:30]: Ideally we'd include a tab for Dart as well, but I can add that after we merge!

File: web/sidebars.js [hunk 2]
```diff
@@ -148,10 +148,12 @@ module.exports = {
             'guides/auth/auth-apple',
             'guides/auth/auth-bitbucket',
             'guides/auth/auth-discord',
+            'guides/auth/auth-email',
```
Comment (kiwicopple) [inline web/sidebars.js:151]: Let's move both of these (email, magic link) to the top of the list since they are the most common

**Review by kiwicopple — CHANGES_REQUESTED** [review]

**Comment by kiwicopple** [comment]
great PR, thanks @Isaiah-Hamilton 🙏

**Comment by Isaiah-Hamilton** [comment]
@kiwicopple I'm a little rusty with Dart so let me know again if you want anything changed.

**Comment by kiwicopple** [comment]
Looks great @Isaiah-Hamilton 🚢 🚢 🚢 🚢

**Review by kiwicopple — APPROVED** [review]

**Pull request merged**